pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -545,7 +551,14 @@ func (d *DependenciesDistributor) removeScheduleResultFromAttachedBindings(bindi
 	for index, binding := range attachedBindings {
 		delete(attachedBindings[index].Labels, bindingLabelKey)
 		updatedSnapshot := deleteBindingFromSnapshot(bindingNamespace, bindingName, attachedBindings[index].Spec.RequiredBy)
+		preserveResourcesOnDeletion, err := obtainPreserveStrategyFromBinding(ctx, d.Client, binding)
```

Comment:
The attached bindings should have the `PreserveResourcesOnDeletion` configuration. Why do we have to fetch it from PP?

Reply:
It may have been overridden by `PreserveResourcesOnDeletion` from dependent resources. Therefore, it needs to be obtained from the PP of the current resource.